pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -210,6 +221,11 @@ func (adc *attachDetachController) Run(stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
 	glog.Infof("Starting Attach Detach Controller")
 
+	if !kcache.WaitForCacheSync(stopCh, adc.podsSynced, adc.nodesSynced, adc.pvcsSynced, adc.pvsSynced) {
```

Comment:
I talked to @pmorie and he thought it would be ok. @kubernetes/sig-storage-pr-reviews are you ok with the attach/detach controller waiting for the shared caches to be populated before it starts processing its queue?